sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala

```diff
@@ -45,21 +46,38 @@ object MetalsPlugin extends AutoPlugin {
       if (in) classDirectory.value
       else semanticdbTargetRoot.value
     },
-    semanticdbOptions ++= {
-      val targetRoot = semanticdbTargetRoot.value
-      val versionOfScala = scalaVersion.value
-      if (ScalaInstance.isDotty(versionOfScala))
-        List("-semanticdb-target", targetRoot.toString)
-      else
-        List(
-          s"-P:semanticdb:sourceroot:${baseDirectory.in(ThisBuild).value}",
-          s"-P:semanticdb:targetroot:$targetRoot",
-          "-Yrangepos",
-          // Needed for "find references" on implicits and `apply` methods.
-          s"-P:semanticdb:synthetics:on",
-          // Don't fail compilation in case of Scalameta crash during SemanticDB generation.
-          s"-P:semanticdb:failures:warning"
-        )
-    }
+    semanticdbOptions ++= createSemanticDbOptions(
+      scalaVersion.value,
+      semanticdbTargetRoot.value,
+      baseDirectory.in(ThisBuild).value
+    )
+  )
+  lazy val testSettings: Seq[Def.Setting[_]] = List(
+    // remove Compile targetRoot from Test config
```

Comment:
Could you maybe comment more about why do we need to remove all semanticdb settings here? Not sure I understand why we need to remove all options here and how they got there in the first place.

Reply:
To be 100% honest, I don't fully get what is going on. I wrestled with this for a while since I was assuming that this is what I wanted:

https://github.com/sbt/sbt/blob/c00196abf0c0b60d301e9516b8513b498359f987/main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala#L75

However, that actually messed things up for me more. So instead I just adapted the other stuff that sbt has, and that ended up working for me. I plan on spending a bit more time on this to get a better understanding because I need to get what's happening here.

Reply:
Actually thinking about this some more, I don't need to remove all of those, especially when it falls into the Scala 2 else. Let me adapt this a bit.

Reply:
So it looks like without our plugin the `semanticdb-target` is already set correctly. So there is no need for us to set it I think.